java/lance-jni/src/blocking_dataset.rs

```diff
@@ -0,0 +1,36 @@
+use arrow::ffi_stream::ArrowArrayStreamReader;
+use tokio::runtime::Runtime;
+
+use lance::Dataset;
+use lance::Result;
+
+pub struct BlockingDataset {
+    inner: Dataset,
+    rt: Runtime,
+}
+
+impl BlockingDataset {
+    pub fn write(reader: ArrowArrayStreamReader, uri: &str) -> Result<Self> {
+        let rt = tokio::runtime::Builder::new_current_thread()
+            .enable_all()
+            .build()?;
+
+        let inner = rt.block_on(
+            Dataset::write(reader, uri, Option::None)
+        )?;
+        Ok(BlockingDataset { inner, rt })
+    }
+    pub fn open(uri: &str) -> Result<Self> {
```

Comment:
it'd be great that we can pass Read / Write options from JNI.

Reply:
Yeah, I'm also thinking about that, I will add the options very soon